### ISSUE TITLE
Change keyboard brightness on Android based on the app's theme

### ISSUE THREAD

**Issue by Michal-MK** [flutter/flutter#159816]
### Steps to reproduce

This is mostly a re-open of #75521

Dismissing it as invalid is wrong based on the fact that for example .NET MAUI is able to do it just fine. Therefore there is in fact a way to control it. even if it may be indirect.

Another fact that makes me think that Flutter already does somehow alter the keyboard brightness on Android, is that when I set my phone to Night mode, the keyboard I use changes in other applications to respect that. But in Flutter the keyboard stays bright. As if Flutter was forcing it to that state.

### Expected results

Brightness.dark should change the keyboard's appearance to dark (given that the keyboard is reasonably smart about it)

### Actual results

The keyboard is bright no matter the app theme, the phone's theme/night mode being enabled

### Code sample

`flutter create --empty` and add a `TextField`

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
fvm flutter doctor -v       
[✓] Flutter (Channel stable, 3.24.3, on macOS 14.3 23D56 darwin-arm64, locale en-CZ)
    • Flutter version 3.24.3 on channel stable at /Users/michalhazdra/fvm/versions/3.24.3
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 2663184aa7 (3 months ago), 2024-09-11 16:27:48 -0500
    • Engine revision 36335019a8
    • Dart version 3.5.3
    • DevTools version 2.37.3

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/xx/Library/Android/sdk
    • Platform android-35, build-tools 34.0.0
    • ANDROID_HOME = /Users/xx/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.9+0-17.0.9b1087.7-11185874)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15E204a
    • CocoaPods version 1.15.2

[✗] Chrome - develop for the web (Cannot find Chrome executable at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[✓] Android Studio (version 2023.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.9+0-17.0.9b1087.7-11185874)

[✓] IntelliJ IDEA Ultimate Edition (version 2023.2.3)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.95.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.102.0

[✓] Connected device (4 available)

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```

</details>

**Comment by darshankawar** [comment]
@Michal-MK 
I think even if you set `theme: ThemeData.dark(),` in your code but if the device's theme is set to light, then the keyboard will be shown in light mode and vice-versa when you set it to `light`. This is what I observed upon running a sample code using latest stable version on S10+ device.

Per https://github.com/flutter/flutter/issues/75521#issuecomment-781887807, it doesn't seem to be specific to Flutter and hence it was closed as out of scope. Also read https://www.reddit.com/r/GooglePixel/comments/u9c7cz/keyboard_not_following_system_theme/?rdt=64138 for further reference and it seems to be Android native discussion more than Flutter's.

**Comment by Michal-MK** [comment]
Yes but when I set my phone to dark mode, and run a Flutter app, the keyboard is light, however, when I for example click on the google search box on the home screen the keyboard is correctly shown as dark.

It is an LG G8s running Android 11. I will try in an emulator and let you know.

**Comment by Michal-MK** [comment]
Well now I tried a new project on the LG, and using ThemeData.light() and having the system be in light mode, the keyboard is dark...

<img width="430" alt="image" src="https://github.com/user-attachments/assets/9a540c5d-fad2-43a1-a66b-1517dd841f58">

However I found out that calling this:
```
  SystemChrome.setSystemUIOverlayStyle(
    const SystemUiOverlayStyle(
      statusBarColor: Colors.transparent,
      statusBarIconBrightness: Brightness.dark,
      systemNavigationBarColor: Colors.white,
    ),
  );
```
Specifically the `systemNavigationBarColor`, when I set it to white, the keyboard is white, and when I set it to black, the keyboard is dark.

So what are the defaults here, because when I was not calling it it was black? And t herefore the keyboard was dark even though the phone was in light mode and the app was in light mode...

**Comment by Michal-MK** [comment]
And when I uninstall the app completely, and comment out the `SystemChrome` call the keyboard reverts to dark theme again...

So I clicked though all the apps I have installed on the phone and majority were in the light theme (as the system is) and majority had white (or bright) navigation bar color. So something in Flutter, or the native project Flutter creates is wrong or not set.

**Comment by darshankawar** [comment]
Thanks for the update. I will keep the issue open based on the behavior reported and for team's tracking on expected behavior although a similar issue was closed https://github.com/flutter/flutter/issues/75521

**Comment by gmackall** [comment]
@Michal-MK from triage: I see you reference the phones global light/dark mode state in a couple of comments, but also reference app themes. Is this issue proposing that the keyboard respond to the global phone state, or the state of a particular app theme?

**Comment by Michal-MK** [comment]
Hi, I would expect to at least respect the device theme, respecting the app theme would be a nice thing to have.
